### PR TITLE
unordered_dense: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **unordered_dense/1.3.1**

unordered_dense seems to be the successor of [robin-hood-hashing](https://github.com/martinus/robin-hood-hashing).

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
Excellent! 👌

**Review by uilianries — DISMISSED** [review]

**Comment by uilianries** [comment]
Ops! I'll need to fix those templates. `target_compile_features` is the correct method.

File: recipes/unordered_dense/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.info.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.get_safe("compiler.version")) < minimum_version:
```
Comment (uilianries) [inline recipes/unordered_dense/all/conanfile.py:51]:
```suggestion
        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
        if minimum_version and Version(self.settings.get_safe("compiler.version")) < minimum_version:
```

It won't be possible to validate, due self.info.clear(). I believe this case can be related to https://github.com/conan-io/conan/issues/12072

Comment (toge) [reply recipes/unordered_dense/all/conanfile.py:51]: @uilianries 
Thanks!
Now I understand.

**Review by uilianries — APPROVED** [review]
LGTM 👌

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**